lib/Backends/NNPI/NNPI.cpp

```diff
@@ -540,11 +542,20 @@ bool NNPIBackend::shouldLower(const Node *N) const {
     return false;
   case Kinded::Kind::SparseLengthsSumNodeKind:
     // WA - lower until ICE-T implements it.
-    if (NNPIBackend::backendOptions_.useIceT ||
-        NNPIBackend::backendOptions_.inferOnDevice) {
-      return true;
+    {
+      const SparseLengthsSumNode *SLS = llvm::cast<SparseLengthsSumNode>(N);
+      if ((NNPIBackend::backendOptions_.useIceT ||
+           NNPIBackend::backendOptions_.inferOnDevice) &&
+          ((SLS->getResult().getElementType() == ElemKind::FloatTy) ||
+           (SLS->getResult().getElementType() == ElemKind::Int8QTy))) {
+        return true;
+      }
     }
     return false;
+  case Kinded::Kind::SwishNodeKind: {
+    NodeInfo NI(*N);
+    return !NI.allInputsAndOutputsHaveSameElemKind({ElemKind::Float16Ty});
+  }
```

Comment:
For all of these, as we discussed we should not lower based on precision inside `shouldLower()`. They should not be lowered here, and the logic for lowering should be moved to `lowerRequiredNodes()` which is called from `transformPostLowering()` after precision transformation has occurred.

Reply:
You are right but I did not fix it yet in this PR since we only just started testing these changes.
Can we keep it this way and fix it in another PR after it is tested?

Reply:
Ok, I may need to change it on our end in our own PR for the Int8 Swish handling. You can pull it in yourself when it makes sense for you.